runintegrationtests.sh

```diff
@@ -39,7 +37,25 @@ touch $FAILURE_TEMP_FILE
 
 cd apis
 
-if [[ "$RETRY_ARG" == "yes" && (-f "$FAILURE_FILE")]]
+if [[ ${#APIS[@]} != 0 ]]
```

Comment:
I think a comment would be useful - it had me confused for a while.
Something like:

```text
# There are three separate ways of determining the tests to run:
# - If APIs have been specified on the command line, test those
# - If --retry has been specified, run the previously-failed tests
# - Otherwise, fine all potential tests
```

Should we also check that `--retry` hasn't been specified along with a set of APIs?

Reply:
I added a check to fail if --retry and projects are set as well as the comment you suggested.